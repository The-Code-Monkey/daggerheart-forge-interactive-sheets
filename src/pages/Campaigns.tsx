
import { useState } from "react";
import { Card, CardContent, CardDescription, CardHeader, CardTitle } from "@/components/ui/card";
import { Button } from "@/components/ui/button";
import { Badge } from "@/components/ui/badge";
import { Users, Calendar, MapPin, Plus, Settings, Eye } from "lucide-react";

const Campaigns = () => {
  const [campaigns] = useState([
    {
      id: 1,
      name: "The Shattered Crown",
      description: "A quest to restore the ancient crown and unite the fractured kingdoms",
      players: 4,
      sessions: 12,
      lastPlayed: "2024-01-15",
      status: "active"
    },
    {
      id: 2,
      name: "Shadows of the Deep",
      description: "Maritime adventures in the mysterious Cerulean Archipelago",
      players: 3,
      sessions: 8,
      lastPlayed: "2024-01-10",
      status: "active"
    },
    {
      id: 3,
      name: "The Dragon's Bargain",
      description: "A completed campaign where heroes made a pact with an ancient dragon",
      players: 5,
      sessions: 24,
      lastPlayed: "2023-12-20",
      status: "completed"
    }
  ]);

  return (
    <div className="min-h-screen bg-gradient-to-b from-purple-900 via-purple-800 to-slate-900 py-8 px-4">
      <div className="max-w-7xl mx-auto">
        {/* Header */}
        <div className="flex flex-col md:flex-row justify-between items-start md:items-center mb-8">
          <div>
            <h1 className="text-4xl font-bold text-white mb-4">Campaign Manager</h1>
            <p className="text-xl text-purple-200">Organize and track your Daggerheart campaigns</p>
          </div>
          <Button className="bg-gradient-to-r from-yellow-500 to-orange-500 hover:from-yellow-600 hover:to-orange-600 text-black font-semibold">
            <Plus className="w-4 h-4 mr-2" />
            New Campaign
          </Button>
        </div>

        {/* Campaign Grid */}
        <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-6 mb-8">
          {campaigns.map((campaign) => (
            <Card key={campaign.id} className="bg-gradient-to-br from-slate-800/80 to-slate-900/80 border-purple-500/30 backdrop-blur-sm hover:border-purple-400/50 transition-all duration-300">
              <CardHeader>
                <div className="flex items-start justify-between">
                  <div>
                    <CardTitle className="text-white text-xl mb-2">{campaign.name}</CardTitle>
                    <Badge
                      variant="secondary"
                      className={campaign.status === 'active' ? 'bg-green-600/50 text-white' : 'bg-gray-600/50 text-white'}
                    >
                      {campaign.status}
                    </Badge>
                  </div>
                  <Button variant="ghost" size="sm" className="text-purple-300 hover:text-white">
                    <Settings className="w-4 h-4" />
                  </Button>
                </div>
                <CardDescription className="text-purple-200">
                  {campaign.description}
                </CardDescription>
              </CardHeader>
              <CardContent className="space-y-4">
                <div className="grid grid-cols-2 gap-4 text-sm">
                  <div className="flex items-center gap-2 text-purple-200">
                    <Users className="w-4 h-4" />
                    {campaign.players} Players
                  </div>
                  <div className="flex items-center gap-2 text-purple-200">
                    <Calendar className="w-4 h-4" />
                    {campaign.sessions} Sessions
                  </div>
                </div>

                <div className="text-sm text-purple-300">
                  Last played: {new Date(campaign.lastPlayed).toLocaleDateString()}
                </div>

                <div className="flex gap-2">
                  <Button variant="outline" className="flex-1 border-purple-400 text-purple-100 ">
                    <Eye className="w-4 h-4 mr-2" />
                    View
                  </Button>
                  <Button variant="outline" className="border-purple-400 text-purple-100 ">
                    <MapPin className="w-4 h-4" />
                  </Button>
                </div>
              </CardContent>
            </Card>
          ))}
        </div>

        {/* Quick Stats */}
        <div className="grid grid-cols-1 md:grid-cols-4 gap-6">
          <Card className="bg-gradient-to-br from-green-800/40 to-emerald-800/40 border-green-500/30 backdrop-blur-sm">
            <CardContent className="p-6 text-center">
              <div className="text-3xl font-bold mb-2 text-green-950">2</div>
              <div className="text-green-950">Active Campaigns</div>
            </CardContent>
          </Card>

          <Card className="bg-gradient-to-br from-blue-800/40 to-cyan-800/40 border-blue-500/30 backdrop-blur-sm">
            <CardContent className="p-6 text-center">
              <div className="text-3xl font-bold text-blue-900 mb-2">12</div>
              <div className="text-blue-900">Total Players</div>
            </CardContent>
          </Card>

          <Card className="bg-gradient-to-br from-purple-800/40 to-pink-800/40 border-purple-500/30 backdrop-blur-sm">
            <CardContent className="p-6 text-center">
              <div className="text-3xl font-bold text-purple-900 mb-2">44</div>
              <div className="text-purple-900">Sessions Played</div>
            </CardContent>
          </Card>

          <Card className="bg-gradient-to-br from-orange-800/40 to-red-800/40 border-orange-500/30 backdrop-blur-sm">
            <CardContent className="p-6 text-center">
              <div className="text-3xl font-bold text-orange-900 mb-2">156</div>
              <div className="text-orange-900">Hours Played</div>
            </CardContent>
          </Card>
        </div>
      </div>
    </div>
  );
};

export default Campaigns;
