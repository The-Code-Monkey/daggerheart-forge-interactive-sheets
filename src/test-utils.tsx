import React from "react";
import { render, RenderOptions } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

// Mock AuthContext
const mockAuthContext = {
  user: null,
  loading: false,
  signIn: jest.fn(),
  signOut: jest.fn(),
  signUp: jest.fn(),
};

const AuthProvider = ({ children }: { children: React.ReactNode }) => {
  return <div data-testid="auth-provider">{children}</div>;
};

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <BrowserRouter>
      <AuthProvider>{children}</AuthProvider>
    </BrowserRouter>
  );
};

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

// Export everything from testing-library/react
export * from "@testing-library/react";
export { customRender as render };

// Mock useAuth hook
export const mockUseAuth = () => mockAuthContext;
