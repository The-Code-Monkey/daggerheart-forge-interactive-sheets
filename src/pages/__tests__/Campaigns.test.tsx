import { render } from "../../test-utils";
import Campaigns from "../Campaigns";

describe("Campaigns Page", () => {
  it("renders correctly", () => {
    const { container } = render(<Campaigns />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
