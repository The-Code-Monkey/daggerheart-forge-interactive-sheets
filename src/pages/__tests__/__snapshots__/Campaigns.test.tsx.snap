// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Campaigns Page renders correctly 1`] = `
<div
  data-testid="auth-provider"
>
  <div
    class="min-h-screen bg-gradient-to-b from-brand-900 via-brand-800 to-slate-900 py-8 px-4"
  >
    <div
      class="max-w-7xl mx-auto"
    >
      <div
        class="flex flex-col md:flex-row justify-between items-start md:items-center mb-8"
      >
        <div>
          <h1
            class="text-4xl font-bold text-white mb-4"
          >
            Campaign Manager
          </h1>
          <p
            class="text-xl text-brand-200"
          >
            Organize and track your Daggerheart campaigns
          </p>
        </div>
        <button
          class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm ring-offset-background transition-colors duration-300 ease-in-out focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 hover:bg-primary/90 h-10 px-4 py-2 bg-gradient-to-r from-yellow-500 to-orange-500 hover:from-yellow-600 hover:to-orange-600 text-black font-semibold"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-plus w-4 h-4 mr-2"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5 12h14"
            />
            <path
              d="M12 5v14"
            />
          </svg>
          New Campaign
        </button>
      </div>
      <div
        class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-6 mb-8"
      >
        <div
          class="rounded-lg border text-card-foreground shadow-sm bg-gradient-to-br from-slate-800/80 to-slate-900/80 border-brand-500/30 backdrop-blur-sm hover:border-brand-400/50 transition-all duration-300"
        >
          <div
            class="flex flex-col space-y-1.5 p-6"
          >
            <div
              class="flex items-start justify-between"
            >
              <div>
                <h3
                  class="font-semibold tracking-tight text-white text-xl mb-2"
                >
                  The Shattered Crown
                </h3>
                <div
                  class="inline-flex items-center rounded-full border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 border-transparent hover:bg-secondary/80 bg-green-600/50 text-white"
                >
                  active
                </div>
              </div>
              <button
                class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium ring-offset-background transition-colors duration-300 ease-in-out focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 hover:bg-accent h-9 rounded-md px-3 text-brand-300 hover:text-white"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-settings w-4 h-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12.22 2h-.44a2 2 0 0 0-2 2v.18a2 2 0 0 1-1 1.73l-.43.25a2 2 0 0 1-2 0l-.15-.08a2 2 0 0 0-2.73.73l-.22.38a2 2 0 0 0 .73 2.73l.15.1a2 2 0 0 1 1 1.72v.51a2 2 0 0 1-1 1.74l-.15.09a2 2 0 0 0-.73 2.73l.22.38a2 2 0 0 0 2.73.73l.15-.08a2 2 0 0 1 2 0l.43.25a2 2 0 0 1 1 1.73V20a2 2 0 0 0 2 2h.44a2 2 0 0 0 2-2v-.18a2 2 0 0 1 1-1.73l.43-.25a2 2 0 0 1 2 0l.15.08a2 2 0 0 0 2.73-.73l.22-.39a2 2 0 0 0-.73-2.73l-.15-.08a2 2 0 0 1-1-1.74v-.5a2 2 0 0 1 1-1.74l.15-.09a2 2 0 0 0 .73-2.73l-.22-.38a2 2 0 0 0-2.73-.73l-.15.08a2 2 0 0 1-2 0l-.43-.25a2 2 0 0 1-1-1.73V4a2 2 0 0 0-2-2z"
                  />
                  <circle
                    cx="12"
                    cy="12"
                    r="3"
                  />
                </svg>
              </button>
            </div>
            <p
              class="text-sm text-brand-200"
            >
              A quest to restore the ancient crown and unite the fractured kingdoms
            </p>
          </div>
          <div
            class="p-6 pt-0 space-y-4"
          >
            <div
              class="grid grid-cols-2 gap-4 text-sm"
            >
              <div
                class="flex items-center gap-2 text-brand-200"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-users w-4 h-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16 21v-2a4 4 0 0 0-4-4H6a4 4 0 0 0-4 4v2"
                  />
                  <path
                    d="M16 3.128a4 4 0 0 1 0 7.744"
                  />
                  <path
                    d="M22 21v-2a4 4 0 0 0-3-3.87"
                  />
                  <circle
                    cx="9"
                    cy="7"
                    r="4"
                  />
                </svg>
                4
                 Players
              </div>
              <div
                class="flex items-center gap-2 text-brand-200"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-calendar w-4 h-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8 2v4"
                  />
                  <path
                    d="M16 2v4"
                  />
                  <rect
                    height="18"
                    rx="2"
                    width="18"
                    x="3"
                    y="4"
                  />
                  <path
                    d="M3 10h18"
                  />
                </svg>
                12
                 Sessions
              </div>
            </div>
            <div
              class="text-sm text-brand-300"
            >
              Last played:
               
              1/15/2024
            </div>
            <div
              class="flex gap-2"
            >
              <button
                class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors duration-300 ease-in-out focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border bg-brand-700/80 hover:bg-brand-700 hover:text-accent-foreground h-10 px-4 py-2 flex-1 border-brand-400 text-brand-100"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-eye w-4 h-4 mr-2"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M2.062 12.348a1 1 0 0 1 0-.696 10.75 10.75 0 0 1 19.876 0 1 1 0 0 1 0 .696 10.75 10.75 0 0 1-19.876 0"
                  />
                  <circle
                    cx="12"
                    cy="12"
                    r="3"
                  />
                </svg>
                View
              </button>
              <button
                class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors duration-300 ease-in-out focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border bg-brand-700/80 hover:bg-brand-700 hover:text-accent-foreground h-10 px-4 py-2 border-brand-400 text-brand-100"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-map-pin w-4 h-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M20 10c0 4.993-5.539 10.193-7.399 11.799a1 1 0 0 1-1.202 0C9.539 20.193 4 14.993 4 10a8 8 0 0 1 16 0"
                  />
                  <circle
                    cx="12"
                    cy="10"
                    r="3"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <div
          class="rounded-lg border text-card-foreground shadow-sm bg-gradient-to-br from-slate-800/80 to-slate-900/80 border-brand-500/30 backdrop-blur-sm hover:border-brand-400/50 transition-all duration-300"
        >
          <div
            class="flex flex-col space-y-1.5 p-6"
          >
            <div
              class="flex items-start justify-between"
            >
              <div>
                <h3
                  class="font-semibold tracking-tight text-white text-xl mb-2"
                >
                  Shadows of the Deep
                </h3>
                <div
                  class="inline-flex items-center rounded-full border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 border-transparent hover:bg-secondary/80 bg-green-600/50 text-white"
                >
                  active
                </div>
              </div>
              <button
                class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium ring-offset-background transition-colors duration-300 ease-in-out focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 hover:bg-accent h-9 rounded-md px-3 text-brand-300 hover:text-white"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-settings w-4 h-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12.22 2h-.44a2 2 0 0 0-2 2v.18a2 2 0 0 1-1 1.73l-.43.25a2 2 0 0 1-2 0l-.15-.08a2 2 0 0 0-2.73.73l-.22.38a2 2 0 0 0 .73 2.73l.15.1a2 2 0 0 1 1 1.72v.51a2 2 0 0 1-1 1.74l-.15.09a2 2 0 0 0-.73 2.73l.22.38a2 2 0 0 0 2.73.73l.15-.08a2 2 0 0 1 2 0l.43.25a2 2 0 0 1 1 1.73V20a2 2 0 0 0 2 2h.44a2 2 0 0 0 2-2v-.18a2 2 0 0 1 1-1.73l.43-.25a2 2 0 0 1 2 0l.15.08a2 2 0 0 0 2.73-.73l.22-.39a2 2 0 0 0-.73-2.73l-.15-.08a2 2 0 0 1-1-1.74v-.5a2 2 0 0 1 1-1.74l.15-.09a2 2 0 0 0 .73-2.73l-.22-.38a2 2 0 0 0-2.73-.73l-.15.08a2 2 0 0 1-2 0l-.43-.25a2 2 0 0 1-1-1.73V4a2 2 0 0 0-2-2z"
                  />
                  <circle
                    cx="12"
                    cy="12"
                    r="3"
                  />
                </svg>
              </button>
            </div>
            <p
              class="text-sm text-brand-200"
            >
              Maritime adventures in the mysterious Cerulean Archipelago
            </p>
          </div>
          <div
            class="p-6 pt-0 space-y-4"
          >
            <div
              class="grid grid-cols-2 gap-4 text-sm"
            >
              <div
                class="flex items-center gap-2 text-brand-200"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-users w-4 h-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16 21v-2a4 4 0 0 0-4-4H6a4 4 0 0 0-4 4v2"
                  />
                  <path
                    d="M16 3.128a4 4 0 0 1 0 7.744"
                  />
                  <path
                    d="M22 21v-2a4 4 0 0 0-3-3.87"
                  />
                  <circle
                    cx="9"
                    cy="7"
                    r="4"
                  />
                </svg>
                3
                 Players
              </div>
              <div
                class="flex items-center gap-2 text-brand-200"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-calendar w-4 h-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8 2v4"
                  />
                  <path
                    d="M16 2v4"
                  />
                  <rect
                    height="18"
                    rx="2"
                    width="18"
                    x="3"
                    y="4"
                  />
                  <path
                    d="M3 10h18"
                  />
                </svg>
                8
                 Sessions
              </div>
            </div>
            <div
              class="text-sm text-brand-300"
            >
              Last played:
               
              1/10/2024
            </div>
            <div
              class="flex gap-2"
            >
              <button
                class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors duration-300 ease-in-out focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border bg-brand-700/80 hover:bg-brand-700 hover:text-accent-foreground h-10 px-4 py-2 flex-1 border-brand-400 text-brand-100"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-eye w-4 h-4 mr-2"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M2.062 12.348a1 1 0 0 1 0-.696 10.75 10.75 0 0 1 19.876 0 1 1 0 0 1 0 .696 10.75 10.75 0 0 1-19.876 0"
                  />
                  <circle
                    cx="12"
                    cy="12"
                    r="3"
                  />
                </svg>
                View
              </button>
              <button
                class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors duration-300 ease-in-out focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border bg-brand-700/80 hover:bg-brand-700 hover:text-accent-foreground h-10 px-4 py-2 border-brand-400 text-brand-100"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-map-pin w-4 h-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M20 10c0 4.993-5.539 10.193-7.399 11.799a1 1 0 0 1-1.202 0C9.539 20.193 4 14.993 4 10a8 8 0 0 1 16 0"
                  />
                  <circle
                    cx="12"
                    cy="10"
                    r="3"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <div
          class="rounded-lg border text-card-foreground shadow-sm bg-gradient-to-br from-slate-800/80 to-slate-900/80 border-brand-500/30 backdrop-blur-sm hover:border-brand-400/50 transition-all duration-300"
        >
          <div
            class="flex flex-col space-y-1.5 p-6"
          >
            <div
              class="flex items-start justify-between"
            >
              <div>
                <h3
                  class="font-semibold tracking-tight text-white text-xl mb-2"
                >
                  The Dragon's Bargain
                </h3>
                <div
                  class="inline-flex items-center rounded-full border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 border-transparent hover:bg-secondary/80 bg-gray-600/50 text-white"
                >
                  completed
                </div>
              </div>
              <button
                class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium ring-offset-background transition-colors duration-300 ease-in-out focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 hover:bg-accent h-9 rounded-md px-3 text-brand-300 hover:text-white"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-settings w-4 h-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12.22 2h-.44a2 2 0 0 0-2 2v.18a2 2 0 0 1-1 1.73l-.43.25a2 2 0 0 1-2 0l-.15-.08a2 2 0 0 0-2.73.73l-.22.38a2 2 0 0 0 .73 2.73l.15.1a2 2 0 0 1 1 1.72v.51a2 2 0 0 1-1 1.74l-.15.09a2 2 0 0 0-.73 2.73l.22.38a2 2 0 0 0 2.73.73l.15-.08a2 2 0 0 1 2 0l.43.25a2 2 0 0 1 1 1.73V20a2 2 0 0 0 2 2h.44a2 2 0 0 0 2-2v-.18a2 2 0 0 1 1-1.73l.43-.25a2 2 0 0 1 2 0l.15.08a2 2 0 0 0 2.73-.73l.22-.39a2 2 0 0 0-.73-2.73l-.15-.08a2 2 0 0 1-1-1.74v-.5a2 2 0 0 1 1-1.74l.15-.09a2 2 0 0 0 .73-2.73l-.22-.38a2 2 0 0 0-2.73-.73l-.15.08a2 2 0 0 1-2 0l-.43-.25a2 2 0 0 1-1-1.73V4a2 2 0 0 0-2-2z"
                  />
                  <circle
                    cx="12"
                    cy="12"
                    r="3"
                  />
                </svg>
              </button>
            </div>
            <p
              class="text-sm text-brand-200"
            >
              A completed campaign where heroes made a pact with an ancient dragon
            </p>
          </div>
          <div
            class="p-6 pt-0 space-y-4"
          >
            <div
              class="grid grid-cols-2 gap-4 text-sm"
            >
              <div
                class="flex items-center gap-2 text-brand-200"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-users w-4 h-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16 21v-2a4 4 0 0 0-4-4H6a4 4 0 0 0-4 4v2"
                  />
                  <path
                    d="M16 3.128a4 4 0 0 1 0 7.744"
                  />
                  <path
                    d="M22 21v-2a4 4 0 0 0-3-3.87"
                  />
                  <circle
                    cx="9"
                    cy="7"
                    r="4"
                  />
                </svg>
                5
                 Players
              </div>
              <div
                class="flex items-center gap-2 text-brand-200"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-calendar w-4 h-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8 2v4"
                  />
                  <path
                    d="M16 2v4"
                  />
                  <rect
                    height="18"
                    rx="2"
                    width="18"
                    x="3"
                    y="4"
                  />
                  <path
                    d="M3 10h18"
                  />
                </svg>
                24
                 Sessions
              </div>
            </div>
            <div
              class="text-sm text-brand-300"
            >
              Last played:
               
              12/20/2023
            </div>
            <div
              class="flex gap-2"
            >
              <button
                class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors duration-300 ease-in-out focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border bg-brand-700/80 hover:bg-brand-700 hover:text-accent-foreground h-10 px-4 py-2 flex-1 border-brand-400 text-brand-100"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-eye w-4 h-4 mr-2"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M2.062 12.348a1 1 0 0 1 0-.696 10.75 10.75 0 0 1 19.876 0 1 1 0 0 1 0 .696 10.75 10.75 0 0 1-19.876 0"
                  />
                  <circle
                    cx="12"
                    cy="12"
                    r="3"
                  />
                </svg>
                View
              </button>
              <button
                class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors duration-300 ease-in-out focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border bg-brand-700/80 hover:bg-brand-700 hover:text-accent-foreground h-10 px-4 py-2 border-brand-400 text-brand-100"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-map-pin w-4 h-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M20 10c0 4.993-5.539 10.193-7.399 11.799a1 1 0 0 1-1.202 0C9.539 20.193 4 14.993 4 10a8 8 0 0 1 16 0"
                  />
                  <circle
                    cx="12"
                    cy="10"
                    r="3"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="grid grid-cols-1 md:grid-cols-4 gap-6"
      >
        <div
          class="rounded-lg border text-card-foreground shadow-sm bg-gradient-to-br from-green-800/40 to-emerald-800/40 border-green-500/30 backdrop-blur-sm"
        >
          <div
            class="p-6 text-center"
          >
            <div
              class="text-3xl font-bold mb-2 text-green-950"
            >
              2
            </div>
            <div
              class="text-green-950"
            >
              Active Campaigns
            </div>
          </div>
        </div>
        <div
          class="rounded-lg border text-card-foreground shadow-sm bg-gradient-to-br from-blue-800/40 to-cyan-800/40 border-blue-500/30 backdrop-blur-sm"
        >
          <div
            class="p-6 text-center"
          >
            <div
              class="text-3xl font-bold text-blue-900 mb-2"
            >
              12
            </div>
            <div
              class="text-blue-900"
            >
              Total Players
            </div>
          </div>
        </div>
        <div
          class="rounded-lg border text-card-foreground shadow-sm bg-gradient-to-br from-brand-800/40 to-pink-800/40 border-brand-500/30 backdrop-blur-sm"
        >
          <div
            class="p-6 text-center"
          >
            <div
              class="text-3xl font-bold text-brand-900 mb-2"
            >
              44
            </div>
            <div
              class="text-brand-900"
            >
              Sessions Played
            </div>
          </div>
        </div>
        <div
          class="rounded-lg border text-card-foreground shadow-sm bg-gradient-to-br from-orange-800/40 to-red-800/40 border-orange-500/30 backdrop-blur-sm"
        >
          <div
            class="p-6 text-center"
          >
            <div
              class="text-3xl font-bold text-orange-900 mb-2"
            >
              156
            </div>
            <div
              class="text-orange-900"
            >
              Hours Played
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
